google-cloud-trace/lib/google/cloud/trace/utils.rb

```diff
@@ -29,8 +29,22 @@ module Utils
         #
         def self.time_to_grpc time
           return nil if time.nil?
+
+          # This is called with different types of objects. Extract
+          # the nanoseconds appropriately.
+          puts "SEPH: ", time.class, time
+          nanos = case
```

Comment:
How about doing something like this instead:

```ruby
time = Time.at(time) if time.is_a? Numeric
```

Reply:
It might make sense to raise `ArgumentError` if the argument is not a supported type.

Reply:
Both of those seem like they'd make this code a lot cleaner. I'll update

Reply:
👍🏼 I prefer this approach.